libvast/vast/system/source.hpp

```diff
@@ -69,14 +69,17 @@ template <class Reader>
 struct source_state {
   // -- member types -----------------------------------------------------------
 
+  using source_actor = caf::stateful_actor<source_state>;
```

Comment:
What's the point of this type alias? Aside from the redundant name `source_state::source_actor`, it doesn't add anything to the state class we can't do already. Seems like you're only using it for `self`, but I see no upside over just using `event_based_actor`.